### PR TITLE
[FLINK-23041][streaming] Added new well defined checkpoint configuration aligned-checkpoint-timeout instead of alignment-timeout

### PR DESCRIPTION
**Pull request by akalash**
## What is the purpose of the change

*This PR adds new configuration aligned-checkpoint-timeout instead of alignment-timeout*

## Brief change log

-*alignment-timeout was deprecated*
-*aligned-checkpoint-timeout was added. It represents time since the global start of the checkpoint after which AC switches to UC*

## Verifying this change

This change added tests and can be verified as follows:

  - *Added  tests in AlternatingCheckpointTest*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (**yes** / no)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (**yes** / no)
  - If yes, how is the feature documented? (not applicable / **docs** / JavaDocs / not documented)

### CODE REVIEW

File: flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/CheckpointConfig.java [hunk 1]
```diff
@@ -783,9 +816,17 @@ public void configure(ReadableConfig configuration) {
         configuration
                 .getOptional(ExecutionCheckpointingOptions.CHECKPOINT_ID_OF_IGNORED_IN_FLIGHT_DATA)
                 .ifPresent(this::setCheckpointIdOfIgnoredInFlightData);
-        configuration
-                .getOptional(ExecutionCheckpointingOptions.ALIGNMENT_TIMEOUT)
-                .ifPresent(this::setAlignmentTimeout);
+
+        Optional<Duration> alignedCheckpointTimeout =
+                configuration.getOptional(ExecutionCheckpointingOptions.ALIGNED_CHECKPOINT_TIMEOUT);
+        if (alignedCheckpointTimeout.isPresent()) {
```
Comment (dawidwys) [inline flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/CheckpointConfig.java:822]: IIRC this whole block is unnecessary. It is already implemented inside of the Configuration if you use the `deprecatedKeys`.

It should be enough to do:
```
Duration alignedCheckpointTimeout = configuration.get(ExecutionCheckpointingOptions.ALIGNED_CHECKPOINT_TIMEOUT);
```

File: flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/io/checkpointing/AlternatingCheckpointsTest.java [hunk 2]
```diff
@@ -353,6 +353,30 @@ public void testTimeoutAlignmentOnFirstBarrier() throws Exception {
...
+                        .build();
+
+        long alignedCheckpointTimeout = 100;
+        // when: Aligned checkpoint timeout expired before the first barrier received.
+        Buffer checkpointBarrier = withTimeout(1, alignedCheckpointTimeout);
+        clock.advanceTime(alignedCheckpointTimeout + 1, TimeUnit.MILLISECONDS);
+
+        ((TestInputChannel) gate.getChannel(0)).read(checkpointBarrier.retainBuffer());
+
+        // then: The UC is triggered as soon as the first barrier will be received.
```
Comment (dawidwys) [inline flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/io/checkpointing/AlternatingCheckpointsTest.java:375]: nit: 
```suggestion
        // then: The UC is triggered as soon as the first barrier is received.
```

File: flink-runtime/src/main/java/org/apache/flink/runtime/jobgraph/tasks/CheckpointCoordinatorConfiguration.java [hunk 3]
```diff
@@ -207,6 +208,7 @@ public int hashCode() {
                 checkpointRetentionPolicy,
                 isExactlyOnce,
                 isUnalignedCheckpointsEnabled,
+                alignedCheckpointTimeout,
```
Comment (dawidwys) [inline flink-runtime/src/main/java/org/apache/flink/runtime/jobgraph/tasks/CheckpointCoordinatorConfiguration.java:211]: Good catch!

File: docs/layouts/shortcodes/generated/rest_v1_dispatcher.html [hunk 4]
```diff
@@ -101,10 +101,10 @@
       "type" : "object",
       "id" : "urn:jsonschema:org:apache:flink:runtime:rest:messages:DashboardConfiguration:Features",
       "properties" : {
-        "web-submit" : {
```
Comment (dawidwys) [inline docs/layouts/shortcodes/generated/rest_v1_dispatcher.html:104]: Out of curiosity. Do you know where does the reordering comes from? Is the file autogenerated? If you don't know don't bother looking for an answer.

**Review by dawidwys — COMMENTED** [review]
Overall looks good. Had one comment to the code. Could you also update the UC documentation page? https://ci.apache.org/projects/flink/flink-docs-master/docs/ops/state/unaligned_checkpoints/ I think adding a sentence or two about what does the timeout mean would be a nice touch.

File: docs/layouts/shortcodes/generated/rest_v1_dispatcher.html (hunk 4 above)
Comment (akalash) [reply docs/layouts/shortcodes/generated/rest_v1_dispatcher.html:104]: This file is autogenerated but it looks like it is generated in alphabet order so it should be correct now. I can only guess but perhaps, initially, the author made these changes manually

**Comment by akalash** [comment]
@dawidwys, I added the documentation - https://github.com/apache/flink/pull/16227/files#diff-330cebde422d7911fd9cb5f32e58770389b13862cb372e6fdb276ba44bb3c4ac, I don't really sure what should I do with the old alignment timeout but as you can see I just deprecated it. I hope it is ok

**Comment by dawidwys** [comment]
Let's just drop the old value. It wasn't properly advertised and I think it doesn't make much sense to document it on that general page.

**Review by dawidwys — APPROVED** [review]
LGTM

**Pull request closed without merging**